Ports/SDL2/patches/0001-Add-SerenityOS-platform-support.patch

```diff
@@ -821,21 +824,46 @@ index 0000000000000000000000000000000000000000..5d0cb527a03dc0ac775ed68d66451495
 +    return 0;
 +}
 +
++static void
++SERENITY_WarpMouse(SDL_Window *window, int x, int y)
++{
++    auto platform_window = SerenityPlatformWindow::from_sdl_window(window);
++
++    auto window_position = platform_window->window()->position();
++    auto warp_position = Gfx::Point(x, y);
++
++    auto absolute_warp_position = window_position + warp_position;
++
++    GUI::ConnectionToWindowServer::the().async_set_global_cursor_position(absolute_warp_position);
++    SDL_SendMouseMotion(window, 0, 0, warp_position.x(), warp_position.y());
++}
++
++static int
++SERENITY_SetRelativeMouseMode(SDL_bool enabled)
++{
++    // FIXME: Implement this properly
++    return 0;
++}
++
 +void
 +SERENITY_InitMouse(_THIS)
 +{
 +    dbgln("SERENITY_InitMouse");
 +
++    // NOTE: By default, SDL uses raw input to implement the relative mouse mode.
++    //       We can tell it to use mouse warping instead since we support mouse warping.
++    SDL_SetHint(SDL_HINT_MOUSE_RELATIVE_MODE_WARP, "1");
```

Comment:
Isn't there something internal that SDL can set? In my experience, hints are usually only used externally.

Reply:
I'm not sure, from what I can see, SDL uses ShouldUseRelativeModeWarp() internally to check for this and all it does is checking if WarpMouse is implemented and if the SDL_HINT_MOUSE_RELATIVE_MODE_WARP is set (see [this](https://github.com/libsdl-org/SDL/blob/031912c4b6c5db80b443f04aa56fec3e4e645153/src/events/SDL_mouse.c#L1058-L1066)). I don't see any other way to set it.

Reply:
I also think the hint should be set for now; the "correct" solution would be to implement a relative mouse mode on the GUI side (as described here: https://wiki.libsdl.org/SDL2/SDL_SetRelativeMouseMode)

Reply:
Actually, looking at [this](https://github.com/libsdl-org/SDL/blob/031912c4b6c5db80b443f04aa56fec3e4e645153/src/events/SDL_mouse.c#L1082-L1090) and testing it, we can remove / return an error code from SetRelativeMouseMode and it will automatically fallback to mouse warping without us having to set the hint. I suppose we could just remove the SetRelativeMouseMode implementation if you prefer then.